cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepository.java

```diff
@@ -223,6 +223,18 @@ public ArtifactDetail getArtifact(Id.Artifact artifactId) throws IOException, Ar
     return artifactStore.getPluginClasses(artifactId, pluginType, pluginName);
   }
 
+  public Map.Entry<ArtifactDescriptor, PluginClass> getPlugin(Id.Artifact artifactId, String pluginType,
```

Comment:
if you just need the location of the plugin's artifact, use getArtifact() rather than adding this method.